src/eth/NonceService.js

```diff
@@ -28,15 +28,25 @@ export default class NonceService extends PublicService {
     }
   }
 
+  _removeDuplicateAddresses(accounts) {
+    const uniqueAddresses = [];
+    accounts.map(account => {
+      if (!uniqueAddresses.includes(account.address))
+        uniqueAddresses.push(account.address);
+    });
+    return uniqueAddresses;
+  }
+
   async setCounts() {
     const accountsList = await this._accountsService.listAccounts();
+    const uniqueAddresses = this._removeDuplicateAddresses(accountsList);
 
     return new Promise(resolve => {
       accountsList.map(async account => {
         const txCount = await this._getTxCount(account.address);
         this._counts[account.address] = txCount;
 
-        if (Object.keys(this._counts).length === accountsList.length) {
+        if (Object.keys(this._counts).length === uniqueAddresses.length) {
```

Comment:
If there are multiple accounts returned in the nonce service with the same address, it throws off the equality comparison and prevents the `setCounts` promise from resolving. By checking for duplicate addresses, and comparing the completed nonce counts with the de-duped list, the equality comparison passes, and the promise resolves.